desc/configuration.py

```diff
@@ -1,17 +1,143 @@
+def change_resolution(x_old, stell_sym, RZ_basis_old, RZ_basis_new, L_basis_old, L_basis_new):
+    """
+
+    Parameters
+    ----------
+    x_old : ndarray
+        DESCRIPTION.
+    RZ_basis_old : FourierZernikeBasis
+        DESCRIPTION.
+    RZ_basis_new : FourierZernikeBasis
+        DESCRIPTION.
+    L_basis_old : DoubleFourierSeries
+        DESCRIPTION.
+    L_basis_new : DoubleFourierSeries
+        DESCRIPTION.
+
+    Returns
+    -------
+    x_new : ndarray
+        
+
+    """
+    old_modes = RZ_basis_old.modes
+    new_modes = RZ_basis_new.modes
+
+    sym_mat_old = symmetry_matrix(RZ_basis_old.modes, L_basis_old.modes, sym=stell_sym)
+    cR_old, cZ_old, cL_old = unpack_state(np.matmul(sym_mat_old, x_old), RZ_basis_old.num_modes)
+
+    cR_new = np.zeros((RZ_basis_new.num_modes,))
+    cZ_new = np.zeros((RZ_basis_new.num_modes,))
+    cL_new = np.zeros((L_basis_new.num_modes,))
+
+    for i in range(RZ_basis_new.num_modes):
+        idx = np.where(np.all(np.array([
+                    np.array(old_modes[:, 0] == new_modes[i, 0]),
+                    np.array(old_modes[:, 1] == new_modes[i, 1]),
+                    np.array(old_modes[:, 2] == new_modes[i, 2])]), axis=0))[0]
+        if len(idx):
+            cR_new[i] = cR_old[idx[0]]
+            cZ_new[i] = cZ_old[idx[0]]
+
+    for i in range(L_basis_new.num_modes):
+        idx = np.where(np.all(np.array([
+                    np.array(old_modes[:, 0] == new_modes[i, 0]),
+                    np.array(old_modes[:, 1] == new_modes[i, 1]),
+                    np.array(old_modes[:, 2] == new_modes[i, 2])]), axis=0))[0]
+        if len(idx):
+            cL_new[i] = cL_old[idx[0]]
+
+    sym_mat_new = symmetry_matrix(RZ_basis_new.modes, L_basis_new.modes, sym=stell_sym)
+    x_new = np.concatenate([cR_new, cZ_new, cL_new])
+    x_new = jnp.matmul(sym_mat_new.T, x_new)
+    return x_new
+
+
+def unpack_state(x, nRZ):
+    """Unpacks the optimization state vector x into cR, cZ, cL components
+
+    Parameters
+    ----------
+    x : ndarray
+        vector to unpack
+    nRZ : int
+        number of R,Z coeffs
+
+    Returns
+    -------
+    cR : ndarray
+        spectral coefficients of R
+    cZ : ndarray
+        spectral coefficients of Z
+    cL : ndarray
+        spectral coefficients of lambda
+
+    """
+
+    cR = x[:nRZ]
+    cZ = x[nRZ:2*nRZ]
+    cL = x[2*nRZ:]
+    return cR, cZ, cL
+
+
 class Configuration():
+    """Configuration constains information about a plasma state, including the 
+       shapes of flux surfaces and profile inputs. It can compute additional 
+       information, such as the magnetic field and plasma currents. 
+    """
 
     # TODO: replace zern_idx & lambda_idx with Transform objects
-    def __init__(self, inputs=None, load_from=None, file_format='hdf5'): #bdry, cP, cI, Psi, NFP, zern_idx, lambda_idx, sym=False, x=None, axis=None) -> None:
+    def __init__(self, inputs:dict=None, load_from:str=None, file_format:str='hdf5') -> None:
```

Comment:
load_from shouldn't be type hinted here: it's supposed to be either a file name OR file instance, so that saving from EquilibriaFamily and Equilibrium can save these without having to open and close the file all the time. I also do type checking in the class, so none is needed at a high-level.